packages/components/src/separator/index.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+export default function Separator( props ) {
+	const { customText, editable, onChange, onKeyDown, className } = props;
+	const inputLength = customText.length + 1;
+
+	const staticLabel = (
+		<span className="components-separator__label">{ customText }</span>
+	);
+	const editableLabel = (
+		<input
+			className="components-separator__label"
+			type="text"
+			value={ customText }
+			size={ inputLength }
+			onChange={ onChange }
+			onKeyDown={ onKeyDown }
+		/>
+	);
+	const label = ( editable ? editableLabel : staticLabel );
```

Comment:
I'm having a hard time saying yes 😁 I think it makes sense in a way, since I can't think of a scenario where you'd want an editable field and not care about when it changes, but conceptually it seems like getting a bit too smart at changing what the component is based on these attributes.

What if the whole label (editable or not) was passed as a child component? We'd still need to create the cross-platform equivalent of [TextInput](https://facebook.github.io/react-native/docs/textinput) and [Text](https://facebook.github.io/react-native/docs/text), but it also means that if there are no children passed, this can also be used for the `HorizontalRule` component we were discussing in #14361